File: app/src/main/java/wang/relish/accumulation/adapter/GoalsAdapter.java
package wang.relish.accumulation.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import wang.relish.accumulation.R;
import wang.relish.accumulation.entity.Goal;

import java.util.ArrayList;
import java.util.List;

/**
 * <pre>
 *     author : 王鑫
 *     e-mail : wangxin@souche.com
 *     time   : 2017/05/04
 *     desc   :
 *     version: 1.0
 * </pre>
 */
public class GoalsAdapter extends BaseAdapter {


    List<Goal> goals = new ArrayList<>();

    public GoalsAdapter(List<Goal> goals) {
        this.goals = goals;
    }

    @Override
    public int getCount() {
        return goals.size();
    }

    @Override
    public Goal getItem(int position) {
        return goals.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        ViewHolder holder;
        if (convertView == null) {
            convertView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_goal, parent, false);
            holder = new ViewHolder(convertView);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }
        holder.tv_name.setText(goals.get(position).getName());
        return convertView;
    }

    private class ViewHolder {
        TextView tv_name;

        ViewHolder(View view) {
            tv_name = (TextView) view.findViewById(R.id.tv_name);
        }
    }
}

File: app/src/main/java/com/qyt/accumulation/adapter/GoalsAdapter.kt
package com.qyt.accumulation.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

import com.qyt.accumulation.R
import com.qyt.accumulation.entity.Goal

import java.util.ArrayList

/**
 * <pre>
 * author : 王鑫
 * e-mail : wangxin@souche.com
 * time   : 2017/05/04
 * desc   :
 * version: 1.0
</pre> *
 */
class GoalsAdapter(goals: List<Goal>) : BaseAdapter() {


    internal var goals: List<Goal> = ArrayList()

    init {
        this.goals = goals
    }

    override fun getCount(): Int {
        return goals.size
    }

    override fun getItem(position: Int): Goal {
        return goals[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView

        val holder: ViewHolder
        if (convertView == null) {
            convertView = LayoutInflater.from(parent.context).inflate(R.layout.item_goal, parent, false)
            holder = ViewHolder(convertView)
            convertView!!.tag = holder
        } else {
            holder = convertView.tag as ViewHolder
        }
        holder.tv_name.text = goals[position].name
        return convertView
    }

    private inner class ViewHolder internal constructor(view: View) {
        internal var tv_name: TextView

        init {
            tv_name = view.findViewById(R.id.tv_name) as TextView
        }
    }
}